lib/src/json_schema/validator.dart

```diff
@@ -444,17 +438,29 @@ class Validator {
         try {
           Uri.parseIPv6Address(instance.data);
         } on FormatException catch (_) {
-          _err('ipv6" format not accepted $instance', instance.path, schema.path);
+          _err('"ipv6" format not accepted $instance', instance.path, schema.path);
         }
         break;
       case 'hostname':
-        if (JsonSchemaValidationRegexes.hostname.firstMatch(instance.data) == null) {
+        final regexp = schema.schemaVersion.compareTo(SchemaVersion.draft2019_09) < 0
+            ? JsonSchemaValidationRegexes.hostname
+            // Updated in Draft 2019-09
+            : JsonSchemaValidationRegexes.hostnameDraft2019;
```

Comment:
`compareTo` is nice and handy when it comes to seeing whether the draft in use is as recent as the draft where some feature was introduced.